src/script/components/LexicalInput/components/Mention/Mention.tsx

```diff
@@ -0,0 +1,212 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+
+import {useLexicalComposerContext} from '@lexical/react/LexicalComposerContext';
+import {useLexicalNodeSelection} from '@lexical/react/useLexicalNodeSelection';
+import {mergeRegister} from '@lexical/utils';
+import {
+  $getNodeByKey,
+  $getSelection,
+  $isDecoratorNode,
+  $isElementNode,
+  $isNodeSelection,
+  $isTextNode,
+  $setSelection,
+  BLUR_COMMAND,
+  CLICK_COMMAND,
+  COMMAND_PRIORITY_LOW,
+  GridSelection,
+  KEY_ARROW_LEFT_COMMAND,
+  KEY_ARROW_RIGHT_COMMAND,
+  KEY_BACKSPACE_COMMAND,
+  KEY_DELETE_COMMAND,
+  NodeKey,
+  NodeSelection,
+  RangeSelection,
+  SELECTION_CHANGE_COMMAND,
+} from 'lexical';
+
+// import {IS_IOS} from './environment';
+import {$isBeautifulMentionNode} from '../../nodes/MentionNode';
+import {getNextSibling, getPreviousSibling} from '../../utils/mention-utils';
+
+interface BeautifulMentionComponentProps {
+  mention: string;
+  nodeKey: NodeKey;
+  className?: string;
+  classNameFocused?: string;
+}
+
+export const Mention = (props: BeautifulMentionComponentProps) => {
+  const {mention, className = '', classNameFocused = '', nodeKey} = props;
+  const [editor] = useLexicalComposerContext();
+  const [isSelected, setSelected, clearSelection] = useLexicalNodeSelection(nodeKey);
+  const [selection, setSelection] = useState<RangeSelection | NodeSelection | GridSelection | null>(null);
+  const isFocused = $isNodeSelection(selection) && isSelected;
+  const ref = useRef(null);
+
+  const classNameFinal = useMemo(() => {
+    const classes = [className];
+    if (isFocused) {
+      classes.push(classNameFocused);
+    }
+    return classes.join(' ').trim() || undefined;
+  }, [className, classNameFocused, isFocused]);
+
+  const onDelete = useCallback(
+    (payload: KeyboardEvent) => {
+      if (isSelected && $isNodeSelection($getSelection())) {
+        payload.preventDefault();
+        const node = $getNodeByKey(nodeKey);
+        if ($isBeautifulMentionNode(node)) {
+          node.remove();
+        }
+        setSelected(false);
+      }
+      return false;
+    },
+    [isSelected, nodeKey, setSelected],
+  );
+
+  const onArrowLeftPress = useCallback(
+    (event: KeyboardEvent) => {
+      const node = $getNodeByKey(nodeKey);
+      if (!node || !node.isSelected()) {
+        return false;
+      }
+      let handled = false;
+      const nodeToSelect = getPreviousSibling(node);
```

Comment:
add proper checks for potential null or undefined values when calling functions like $getNodeByKey(nodeKey) and $isElementNode, $isTextNode, and $isDecoratorNode. 

```
if (nodeToSelect) {
   if ($isElementNode(nodeToSelect)) {
  // code here
  }
} else {
    node.selectPrevious();
    handled = true;
}
```